server/db/pinSchema.js

```diff
@@ -21,26 +21,22 @@ const pinSchema = new mongoose.Schema(
   { strict: 'throw' }
 )
 
-const Pin = new mongoose.model('Pin', pinSchema)
+class PinClass {
+  static async findandRemoveOldPins() {
+    const ttl = process.env.TTL || 14
+    let cutoffDate = new Date()
+    cutoffDate.setDate(cutoffDate.getDate() - ttl)
 
-const findandRemoveOldPins = async () => {
-  const ttl = process.env.TTL || 14
-  let cutoffDate = new Date()
-  cutoffDate.setDate(cutoffDate.getDate() - ttl)
+    const oldPins = await this.find({ time: { $lt: cutoffDate } }).exec()
+    await Promise.all(
+      oldPins.map(async pin => {
+        await ipfs.node.pin.rm(pin.cid)
```

Comment:
Not totally sure about this one, but I _think_ you have to return a promise here. All you have to do is remove the curly braces